content/docs/command-reference/remote/modify.md

```diff
@@ -310,19 +310,99 @@ $ dvc remote modify myremote endpointurl \
   ```
 
-> The connection string contains sensitive user info. Therefore, it's safer to
-> add it with the `--local` option, so it's written to a Git-ignored config
-> file.
+  > The connection string contains sensitive user info. Therefore, it's safer to
+  > add it with the `--local` option, so it's written to a Git-ignored config
+  > file.
 
-For more information on configuring Azure Storage connection strings, visit
-[here](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string).
+  For more information on configuring Azure Storage connection strings, visit
+  [here](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string).
+
+* `account_name` - storage account name, will be required if authenticated
+  either with `account_key` or `sas_token`.
+
+  ```dvc
+  $ dvc remote modify --local myremote account_name \
+                              'account_name'
+  ```
+
+* `account_key` - storage account key:
+
+  ```dvc
+  $ dvc remote modify --local myremote account_key \
+                              'account_key'
+  ```
+
+* `sas_token` - shared access signature token, can be used instead of
+  `account_key`:
+
+  ```dvc
+  $ dvc remote modify --local myremote sas_token \
+                              'sas_token'
+  ```
+
+* `tenant_id` - Tenant ID for
+  [Principal Service Authentication](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+  ```dvc
+  $ dvc remote modify --local myremote tenant_id \
+                              'tenant_id'
+  ```
+
+* `client_id` - Client ID for
+  [Principal Service Authentication](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+  ```dvc
+  $ dvc remote modify --local myremote client_id \
+                              'client_id'
+  ```
+
+* `client_secret` - Client Secret for
+  [Principal Service Authentication](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+  ```dvc
+  $ dvc remote modify --local myremote client_secret \
+                              'client_secret'
+  ```
+
+Azure remotes can also be configured entirely via environment variables:
+
+```dvc
+$ export AZURE_STORAGE_ACCOUNT_NAME='account_name'
+$ export AZURE_STORAGE_ACCOUNT_KEY='account_key'
+$ dvc remote add -d myremote azure://mycontainer/path
+```
+
+For using service principal with certificate:
+
+```dvc
+$ export AZURE_TENANT_ID='tenant_id'
+$ export AZURE_CLIENT_ID='client_id'
+$ export AZURE_CLIENT_CERTIFICATE_PATH='/path/to/certificate'
+```
+
+For logging in with username/password;
+
+```
+$ export AZURE_CLIENT_ID='client_id'
+$ export AZURE_USERNAME='username'
+$ export AZURE_PASSWOR='password'
+```
+
+> On windows, if none of the above settings are configured the login will
+> fallback to search for a signed in Microsoft application (e.g Visual Studio)
```

Comment:
I thinks we should expand and move this description to the top - describe the flow first on different systems, etc

Reply:
for example, have we made changes to support CLI login (default credentials)?

Reply:
> have we made changes to support CLI login

What do you mean by that?

Reply:
Yeah I think we can add some sort of list to show the order

```
The order the authentication options will be tried;
- connection_string
- principals
- account name / account key
- ...
```

Reply:
I meant the DefaultCredentials flow that I shared the link to. `https://github.com/Azure/azure-sdk-for-python/blob/09521dfb517e0859ec961cae006fb728d787b565/sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py#L33` - is part of it (I'm not sure btw, if it is covered by us parsing the config file as we do)

Reply:
no, default credentials flow happens only if you don't set anything in your config.

Reply:
@isidentical What happens first, default credentials or Microsoft account login?

Reply:
> What happens first, default credentials or Microsoft account login?

What do you mean by Microsoft account login? Default credentials flow happens always the last thing if you don't present anything in your config (even the account_name).

Reply:
I was referring to "If no params are given, DVC will try to use a [default credential](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential) (inferred from environment variables)."

Reading that article, I'm guessing they're the same thing?

Reply:
> inferred from environment variables

This is quite confusing combined with that link. The article lists 5 possible id sources, of which only the first one involves env vars. What do we mean exactly? Thanks